src/v1/Models/Item.php

```diff
@@ -46,6 +48,23 @@ class Item extends Model
     // 'getItems'
   ];
 
+  public static function boot()
+  {
+    parent::boot();
+    static::creating(function ($model)
+    {
+      /**
+       * Manage id_bytype field
+       * Get max id in DB for this type and increase 1 to get new id_bytype
+       */
+      $subquery = DB::table((new self())->getTable() . ' as item_alias')
```

Comment:
(nitpicking) `item_alias` seems longer than the initial table name (`items`) :p

Reply:
it's a subquery, so can't use same name table, that's why I use an alias. You problem is with the size of the alias name?

Reply:
Not a problem at all, but I didn't see the point of using an alias here, especially if it's longer than the initial table name. By the way, the alias is not used in your query, shouldn't be used with `type_id`?

Reply:
Ah yes good point, will try it ;)